File: Java/Strings/Count all position in string.java
//Cach 1 Dem cac vi tri cua ki tụ do (Phan biet chu thuong va hoa)
import java.util.Scanner;

public class Array {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next(); 
        char c = sc.next().charAt(0); 

        StringBuilder positions = new StringBuilder(); // To store positions of the character

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == c) {
                positions.append(i).append(" "); // Append the index to the positions StringBuilder
            }
        }

        if (positions.length() > 0) {
            System.out.println(positions.toString().trim()); // Print all positions found
        } else {
            System.out.println("-1"); // Print -1 if the character is not found
        }
        
        sc.close();
    }
}

//Cach 2: Dem cac vi tri cua ki tụ do (Khong ke chu hoa hay thuong)
import java.util.Scanner;

public class Array {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();  // Read input string
        char c = sc.next().charAt(0);  // Read input character

        // Convert both the string and the character to lower case for case-insensitive comparison
        s = s.toLowerCase();
        c = Character.toLowerCase(c);

        String positions = "";  // To store positions of the character in the string
        boolean found = false;  // To track if the character is found

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == c) {  // Compare lower case character
                positions += i + " ";  // Append position to the result string
                found = true;  // Mark that at least one character has been found
            }
        }

        // Print the result based on whether the character was found
        if (found) {
            System.out.print("Character found at positions: " + positions.trim());
        } else {
            System.out.print("-1");  // If no character is found, print -1
        }

        sc.close();
    }
}

File: C/String/Count all position in string.c
//Cach 1 Dem cac vi tri cua ki tụ do (Phan biet chu thuong va hoa)
#include <stdio.h>
#include <string.h>

int main() {
    char c;
    char s[100];

    // Read character and string input
    scanf("%c", &c);
    scanf("%s", s);

    int len = strlen(s);
    int found = 0; // Flag to track if the character is found

    // Loop through the string to find all occurrences of the character 'c'
    for (int i = 0; i < len; i++) {
        if (s[i] == c) {
            printf("%d ", i);  // Print the index of each occurrence
            found = 1;         // Set flag to indicate character found
        }
    }

    // If the character was not found in the string, print -1
    if (!found) {
        printf("-1");
    }

    return 0;
}

//Cach 2: Dem cac vi tri cua ki tụ do (Khong ke chu hoa hay thuong)
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main() {
    char c;
    char s[100];
    scanf("%c", &c);
    scanf("%s", s);

    // Convert the input character to both lower and upper cases
    char lower_c = tolower(c);
    char upper_c = toupper(c);

    int len = strlen(s);
    int found = 0; // To keep track of whether we found the character or not

    for (int i = 0; i < len; i++) {
        // Check for both lower and upper case occurrences of c
        if (s[i] == lower_c || s[i] == upper_c) {
            printf("Found at index: %d\n", i);
            found = 1;
        }
    }

    if (!found) {
        printf("Character not found\n");
    }

    return 0;
}

